python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -329,17 +345,18 @@ async def _process_tool_calls(
         result: Union[OpenAIChatMessageContent, OpenAIStreamingChatMessageContent],
         kernel: "Kernel",
         chat_history: ChatHistory,
+        arguments: "KernelArguments",
     ) -> None:
         """Processes the tool calls in the result and return it as part of the chat history."""
         logger.info(f"processing {len(result.tool_calls)} tool calls")
+        args_cloned = copy(arguments)
         for tool_call in result.tool_calls:
             if tool_call.function is None:
                 continue
             logger.info(f"Calling {tool_call.function.name} function with args: {tool_call.function.arguments}")
             try:
-                func_result = await kernel.invoke(
-                    **tool_call.function.split_name_dict(), arguments=tool_call.function.to_kernel_arguments()
-                )
+                args_cloned.update(tool_call.function.parse_arguments())
```

Comment:
might be good to have parse_arguments in a seperate try except block, or to add a specific except for the one returned by parse_args, with auto invoke that can then be fed back to the model.